honeybee_energy/config.py

```diff
@@ -0,0 +1,322 @@
+"""Honeybee_energy configurations.
+
+Import this into every module where access configurations are needed.
+
+Usage:
+    from honeybee_energy.config import folders
+    print(folders.energyplus_path)
+    print(folders.openstudio_path)
+    folders.energyplus_path = "C:/EnergyPlusV9-0-1"
+"""
+import os
+import sys
+import json
+
+
+class Folders(object):
+    """Honeybee_energy folders.
+
+    Args:
+        mute: If False, the paths to the various folders will be printed as they
+            are found. If True, no printing will occur upon initialization of this
+            class. Default: True.
+
+    Properties:
+        * openstudio_path
+        * energyplus_path
+        * template_library_folder
+        * default_simulation_folder
+        * energy_model_measure_path
+        * python_package_path
+        * python_exe_path
+
+    Usage:
+        folders = Folders(mute=False)
+        print(folders.energyplus_path)
+    """
+
+    def __init__(self, mute=True):
+        # set the mute value
+        self.mute = bool(mute)
+
+        # load paths from the config JSON file within this package
+        config_file = os.path.join(os.path.dirname(__file__), 'config.json')
```

Comment:
I know this is how config.json is implemented in honeybee but there are a couple of issues with this implementation. The most important one is that the user values will be overwritten every time that the library is updated.

I think at the minimum we should write config_file as a property so one can change it to a different file. This will be useful for our plugins in-which we probably want to create the file in the installation folder and not inside the Python libraries installation folder.

Something else to think about is if we want to have a config file for honeybee and each extension (e.g. honeybee-energy) extends the config file? With that approach folders will be modified from a single place. We can also add a component to the interface to set these values for each extension.

Reply:
I was thinking about this case when I put together this class and you can see that the publicly-available `load_from_file` method in this PR addresses this exact concern. We could also do this as a property that sets all of the other properties of the class but, at that point, it feels like it really should be a method as I have now.

And I hear you that there may be cases where you don't want to overwrite the config file but, at the least, I think you need a config.json file in the python package to catch the case that we don't find a config.json file in the user folder.

Reply:
That's a good option to have but `load_from_file` ideally should be part of the setter. Right now one might set the config file without calling load_from_file and that won't trigger loading the files.

``` python
@property 
def config_file(self):
    return self._config_file

@config_file.setter
def config_file(self, cfg):
    # extra checks
    self._load_from_file(cfg)
    # or something similar to this
```

Reply:
I see. So you think there are cases where we will want to check the location of the `config_file` being used. Ok. I'll make it a set-able property and make `load_from_file` hidden.

Reply:
Done.